.travis.yml

```diff
@@ -57,8 +57,9 @@ script:
     cd graphqlapi/prototype
     # install deps
     npm install
-    # run nodemon in the background
+    # run nodemon in the background - sleep to wait for startup of node server
     nohup node prototype-server.js &
+    sleep 15
```

Comment:
@moretea or @etiennedi does one of you guys know how to do this more elegantly? Travis needs to wait for `nohup node prototype-server.js &` to finish.

Reply:
Yes, check out what I did with weaviate. I have written a [shell script](https://github.com/creativesoftwarefdn/weaviate/blob/develop/tools/test/run_ci_server.sh) used [here](https://github.com/creativesoftwarefdn/weaviate/blob/develop/.travis.yml#L36)  that uses docker-compose to start weaviate and it's dependencies. The script then polls & waits until Weaviate is ready.

Reply:
if it should finish anyway, couldn't you just remove both the `nohup` and the `&`? why do you have to run it in a background process in this case?

Reply:
> Yes, check out what I did with weaviate. I have written a [shell script](https://github.com/creativesoftwarefdn/weaviate/blob/develop/tools/test/run_ci_server.sh) used [here](https://github.com/creativesoftwarefdn/weaviate/blob/develop/.travis.yml#L36) that uses docker-compose to start weaviate and it's dependencies. The script then polls & waits until Weaviate is ready.

Thanks, @moretea that's what I was looking for 👍 

> if it should finish anyway, couldn't you just remove both the `nohup` and the `&`? why do you have to run it in a background process in this case?

In this case, it is a daemon @etiennedi, and because vanilla node doesn't support running as a daemon, ⬆️ is faster and more practical IMHO.